code/game/objects/items/melee/misc.dm

```diff
@@ -326,6 +334,201 @@
 	cooldown = 10
 	stamina_damage = 20
 
+//Former Wooden Baton
+/obj/item/melee/classic_baton/police/beater
+	name = "Police Tonfa"
+	desc = "Favored by hot headed Security Officers who don't want to get in trouble with CentCom but still want to get that nostalgic feeling of beating some criminal scum upside the head with a chunk of wood."
+	icon_state = "beater"
+	item_state = "beater"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	force = 8
+	throwforce = 7
+	cooldown = 0
+	stamina_damage = 17
+	stun_animation = TRUE
+	/// Per-mob sleep cooldowns.
+	/// [mob] = [world.time where the cooldown ends]
+	var/static/list/sleep_cooldowns = list()
+	/// Per-mob trip cooldowns.
+	/// [mob] = [world.time where the cooldown ends]
+	var/static/list/trip_cooldowns = list()
+
+/obj/item/melee/classic_baton/police/beater/attack(mob/living/target, mob/living/user)
+	if(!on)
+		return ..()
+	var/def_check = target.getarmor(type = MELEE, penetration = armour_penetration)
+
+	add_fingerprint(user)
+	if((HAS_TRAIT(user, TRAIT_CLUMSY)) && prob(50))
+		to_chat(user, "<span class ='danger'>You hit yourself over the head.</span>")
+		user.adjustStaminaLoss(stamina_damage)
+
+		additional_effects_carbon(user) // user is the target here
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.apply_damage(2*force, BRUTE, BODY_ZONE_HEAD)
+		else
+			user.take_bodypart_damage(2*force)
+		return
+	if(iscyborg(target))
+		// We don't stun if we're on harm.
+		if (user.a_intent != INTENT_HARM)
+			if (affect_silicon)
+				var/list/desc = get_silicon_stun_description(target, user)
+
+				target.flash_act(affect_silicon = TRUE)
+				target.Paralyze(stun_time_silicon)
+				additional_effects_silicon(target, user)
+
+				user.visible_message(desc["visible"], desc["local"])
+				playsound(get_turf(src), on_stun_sound, 100, TRUE, -1)
+
+				if (stun_animation)
+					user.do_attack_animation(target)
+			else
+				..()
+		else
+			..()
+		return
+	if(!isliving(target))
+		return
+	if (user.a_intent == INTENT_HARM)
+		if(!..())
+			return
+		if(!iscyborg(target))
+			return
+	else
+		if(COOLDOWN_FINISHED(src, cooldown_check))
+			if(ishuman(target))
```

Comment:
this proc has issues with massive `if, else` chains.

I see you are using early returns, but maybe not enough or not in the right way.
Try to minimize indentation while maintaining the same logic.

If needed, you can also break parts into their own procs to reduce the indention level. You should not get over 2 indentation for the 'main code', otherwise the code is nigh unreadable.

Reply:
Yeah I'll probably need to redo all of this since I learned a lot since this, it was my first PR!

Reply:
I solved a bit i think

Reply:
I tried doing my absolute best, but I was unable to create procs for the limb damage.
I would need external help to do this as I am unable. So for now, this remains and if someone has something to say I'll have to request assistance